hare/xfailure_test.go

```diff
@@ -0,0 +1,332 @@
+package hare
```

Comment:
The name of this file is not great. It tests both failures and successes.

At first I thought you missed writing success tests as I reviewed this file last. It might be confusing for someone looking for a test in the future.